pkg/sink/cloudstorage/path.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudstorage
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pingcap/tidb/br/pkg/storage"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/engine/pkg/clock"
+	"github.com/pingcap/tiflow/pkg/config"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+)
+
+const (
+	// 3 is the length of "CDC", and the file number contains
+	// at least 6 digits (e.g. CDC000001.csv).
+	minFileNamePrefixLen = 9
+	defaultIndexFileName = "CDC.index"
+)
+
+// GenerateSchemaFilePath generates schema file path based on the table definition.
+func GenerateSchemaFilePath(def TableDefinition) string {
+	return fmt.Sprintf("%s/%s/%d/schema.json", def.Schema, def.Table, def.TableVersion)
+}
+
+type indexWithDate struct {
+	index              uint64
+	currDate, prevDate string
+}
+
+// VersionedTable is used to wrap TableName with a version.
+type VersionedTable struct {
+	model.TableName
+	Version uint64
```

Comment:
Please add a comment to explain the meaning of version.

Reply:
done